core/services/pipeline/runner.go

```diff
@@ -220,40 +244,81 @@ func (r *runner) ExecuteRun(
 
 					t := time.Now()
 					scheduler.report(todo, TaskRunResult{
+						ID:         uuid.NewV4(),
 						Task:       taskRun.task,
 						Result:     Result{Error: ErrRunPanicked{err}},
-						FinishedAt: t,
+						FinishedAt: &t,
 						CreatedAt:  t, // TODO: more accurate start time
 					})
 				}
 			}()
-			result := r.executeTaskRun(ctx, spec, taskRun, l)
+			result := r.executeTaskRun(ctx, run.PipelineSpec, taskRun, l)
 
-			logTaskRunToPrometheus(result, spec)
+			logTaskRunToPrometheus(result, run.PipelineSpec)
 
 			scheduler.report(todo, result)
 		}(taskRun)
 	}
 
-	finishRun := time.Now()
-	runTime := finishRun.Sub(startRun)
-	l.Debugw("Finished all tasks for pipeline run", "specID", spec.ID, "runTime", runTime)
-	PromPipelineRunTotalTimeToCompletion.WithLabelValues(fmt.Sprintf("%d", spec.JobID), spec.JobName).Set(float64(runTime))
+	// if the run is suspended, awaiting resumption
+	run.Pending = scheduler.pending
+	run.State = RunStatusSuspended
 
-	var taskRunResults TaskRunResults
+	if !scheduler.pending {
+		now := time.Now()
+		run.FinishedAt = &now
+
+		// TODO: runTime can be very long now because it'll include suspend
```

Comment:
This is a bit harder to do because the task could be suspended several times. We would need to keep a duration field and accumulate the runtime into it. Should I do that?